### ISSUE TITLE
Parallel test execution

### ISSUE THREAD

**Issue by levBagryansky** [objectionary/eo#1996]
According to [Parallel execution title](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parallel-execution) and [Configuration parameters](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven-config-params) there is necessary to configure parallel execution in pom.xml to run test in parallel.
I added 
```
          <properties>
            <configurationParameters>
              junit.jupiter.execution.parallel.enabled = true
            </configurationParameters>
          </properties>
```
to `maven-surefire-plugin` in `eo-maven-plugin/pom.xml` so testing time reduced by more than 10 sec. That is why I suspect tests with annotation `@Execution(ExecutionMode.CONCURRENT)` are acctually executed by one thread.
@volodya-lombrozo Are we sure they are executed parallely?

**Comment by volodya-lombrozo** [comment]
@levBagryansky Thank you for the issue. Could you add more details why are you suspect that tests are executed in one thread?

**Comment by levBagryansky** [comment]
@volodya-lombrozo I did not find the `junit.jupiter.execution.parallel.enabled = true` option that enables parallel execution in `pom.xml`. Also execution time has been reduced aster adding this option on my laptop.

**Comment by volodya-lombrozo** [comment]
@levBagryansky I see your point here. Yes, in order to run tests in parallel we have to do 2 things:
1. Add `junit.jupiter.execution.parallel.enabled = true` property 
2. Add `@Execution(ExecutionMode.CONCURRENT)` annotation

Let's do it. Apparently it will increase the speed of the build significantly

**Comment by 0pdd** [comment]
@levBagryansky 4 puzzles [#2014](https://github.com/objectionary/eo/issues/2014), [#2015](https://github.com/objectionary/eo/issues/2015), [#2016](https://github.com/objectionary/eo/issues/2016), [#2017](https://github.com/objectionary/eo/issues/2017) are still not solved.

**Comment by 0pdd** [comment]
@levBagryansky the puzzle [#2014](https://github.com/objectionary/eo/issues/2014) is still not solved; solved: [#2015](https://github.com/objectionary/eo/issues/2015), [#2016](https://github.com/objectionary/eo/issues/2016), [#2017](https://github.com/objectionary/eo/issues/2017).

**Comment by 0pdd** [comment]
@levBagryansky all 4 puzzles are solved here: [#2014](https://github.com/objectionary/eo/issues/2014), [#2015](https://github.com/objectionary/eo/issues/2015), [#2016](https://github.com/objectionary/eo/issues/2016), [#2017](https://github.com/objectionary/eo/issues/2017).
